src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs

```diff
@@ -27,6 +29,20 @@ public static bool TryGetRule(this Run run, string ruleId, string ruleKey, out I
                     rule = run.Rules[ruleId];
                 }
             }
+            // No rule in log file, created "fake" rule
+            else
+            {
+                Byte[] ruleLookupBytes = Resource1.ruleLookup;
+                string ruleLookupText = Encoding.UTF8.GetString(ruleLookupBytes);
+                JObject metadata = JObject.Parse(ruleLookupText);
+
+                if (metadata[ruleId] != null)
+                {
+                    string ruleName = metadata[ruleId]["heading"].Value<string>();
```

Comment:
metadata[ruleId]["heading"] could be null. We should add a null check before using it.